### ISSUE TITLE
IL2CPP Login Issues

### ISSUE THREAD

**Issue by Sunndavar** [Microsoft/xbox-live-unity-plugin#204]
There seems to be an issue which is related to the scripting backend IL2CPP.

If I build my project with .net as scripting backend everything works correctly and if I test within the editor I also see the fake user signed in. 

Once I have built the solution with the IL2CPP backend I then add the XBoxServices.config to the project manually (because it doesn't do so automatically on IL2CPP - please correct me if I am mistaken here). I set it to Content=yes, FileType=Document, Include In Projects=True and ItemType=Text (in properties).     

Building the VS Project at this point then throws an exception in the file RuntimeType.cpp line 72. which says IL2CPP_UNREACHABLE; which seems confusing as I don't get this error as long as the XBoxServices.config is not added to the project. I am also seeing the error messages "Invalid token Text at root level document" and "XML document must contain a root level element" which is obviously referring to the XBoxServices.config and I guess is trying to interpret it as an XML file while it is an JSON. 

Please let me know if I am doing something wrong or somehow missed a step in the setup of IL2CPP or if this is an actual bug of the plugin.

I am using Visual Studio 2017, the latest XBoxLive sdk and Unity2017.2.1p1. The Runtime Version and the API Compatibility Level in Unity is set to .NET 4.6.

**Comment by jplafonta** [comment]
The config file should be added to the project automatically. We have a post processing script that modifies the resulting vcxproj file to add the config file; you are welcome to debug into that script (XboxLivePostProcessing.cs) yourself, but I will also try to take a look and see if I can reproduce what you are seeing.

**Comment by Sunndavar** [comment]
I have now updated to the latest XBox Sdk and Unity version but the bug is still the same.

I have also tried to modify the XBoxLivePostProcessing.cs and tested some more. Here are the results:

Normal XBoxLivePostProcessing.cs in Unity:
- Unity throws an build error stating that the .csproj could not be found. However it still creates a VS project. 
- It does copy XBoxServices.config to the folder but does not add it to the solution. Building the project now (in VS) results in an Unity Error in development console: FileNotFoundExeption: Unable to find Xbox live app configuration file “XBoxServices.config” 
- Adding the .config file manually to the solution and changing “Content” to “True” breaks at every IL2CPP_UNREACHABLE in the code and also shows the XML errors from the last post.

Changing the project name in XBoxLivePostProcessing.cs in Unity (.csproj to .vcxproj):
- The FileNotFound build error in unity disappears.
- The .config file is still not added to the solution automatically.
- Adding it manually still causes VS to show the XML error which is described in the last post.
- Building the app without adding the config file manually does not show any errors but also does not show the login in screen of xbox live.
- Adding the config file manually and then building still causes the code to break at every IL2CPP_UNREACHABLE within the code.

Using the .Net scripting backend everything still works perfectly fine.

**Comment by Sunndavar** [comment]
Addition:
The XML error might not be an actual issue as I just recognized that the same XML error is shown in the .Net build, which is working just fine.

**Comment by jplafonta** [comment]
Hello @Sunndavar 

I am a little confused about your statement that the config file is not getting added.  In the latest release we added a check for il2cpp to modify the vcxproj (See [here](https://github.com/Microsoft/xbox-live-unity-plugin/blob/0598d90033e4f9f6e612c1e508e5729d5a429cad/Assets/Xbox%20Live/Editor/XboxLivePostProcessing.cs#L112)).

I've just tested with our sample social scene on Unity versions 2017.3.1f1 and 2017.2.0p2 and both of them generate a working UWP output project with 1802 plugin release.  If you are using something other than the 1802 release, il2cpp is not supported.

**Comment by Sunndavar** [comment]
Hi jplafonta, 

thanks for the answer. Actually your question pretty much pointed out the issue itself. I used the download link under getting started form here: https://github.com/Microsoft/xbox-live-unity-plugin which leads to version 1711 instead of the latest version 1802 therefore I falsely assumed to have the latest version, which obviously I did not have. Therefore, its all my fault ;) You guys might wanna update the link to the latest version so that no-one else is doing my mistake. 

Thanks for your help. After updating everything worked just as expected.
